unit-tests/src/test/java/org/eclipse/collections/impl/bimap/mutable/HashBiMapTest.java

```diff
@@ -102,4 +105,23 @@ public void withKeysValues()
         Assert.assertSame(map, map44);
         Assert.assertSame(map3, map4);
     }
+
+    @Test
+    public void forcePut_inverseKeyPreservation()
+    {
+        Key key1 = new Key("1");
+        Key key2 = new Key("2");
+
+        Key value1 = new Key("abc");
+        Key value2 = new Key("xyz");
+
```

Comment:
No need to inline it. What I mean is that the `Key` type exists to be explicit about `Strings` that are equal but not the same instance. You have code to create a duplicate key and later assert you find the same instance.

```java
Key duplicateOfKey1 = new Key("1");
...
Assert.assertSame(key1, Iterate.getFirst(biMap.entrySet()).getKey());
```

But there's no code like that for values. If it's important to assert separately, then you're missing assertions about values (or the inverse bimap's keys). If it's not important to test separately, then it's unnecessary to use `BiMap<Key, Key>` and you could simplify to `BiMap<Key, String>`.